src/compas_viewer/layout/propertyform.py

```diff
@@ -0,0 +1,425 @@
+        """Add a label to the layout.
+
+        Parameters
+        ----------
+        text : str
+            Text of the label.
+        layout : :PySide6:`PySide6/QtWidgets/QLayout`
+            Layout to be added to.
+
+        Returns
+        -------
+        label : :PySide6:`PySide6/QtWidgets/QLabel`
+            The label added to the layout.
+
+        """
+        label = QLabel(str(text))
+        layout.addWidget(label)
+        return label
+
+    def add_collapsiblebox(self, name: str, layout: QLayout) -> CollapsibleBox:
+        """Add a collapsible box to the layout.
+
+        Parameters
+        ----------
+        name : str
+            Name of the collapsible box.
+        layout : :PySide6:`PySide6/QtWidgets/QLayout`
+            Layout to be added to.
+
+        Returns
+        -------
+        cb : :class:`compas_viewer.layout.CollapsibleBox`
+            The collapsible box added to the layout.
+        """
+        cb = CollapsibleBox(name=name)
+        layout.addWidget(cb)
+        return cb
+
+    def add_transformation(self, obj: "ViewerSceneObject", layout: QLayout):
+        """Map the transformation of an object.
+
+        Parameters
+        ----------
+        obj : :class:`compas_viewer.scene.ViewerSceneObject`
+            The object to be edited.
+        layout : :PySide6:`PySide6/QtWidgets/QLayout`
+            Layout to be added to.
+
+        Returns
+        -------
+        collapsibleBox : :class:`compas_viewer.layout.CollapsibleBox`
+            The collapsible box added to the layout.
+        """
+
+        def set_transformation(value: float, label: str, index: int):
+            if obj.transformation is None:
+                obj.transformation = Transformation()
+            s, _, r, t, _ = obj.transformation.decomposed()
+            if label == "Translation":
+                _vectors = t.translation_vector
+                _vectors[index] = value
+                t = Translation.from_vector(_vectors)
+            elif label == "Scale":
+                _scales = [s[0, 0], s[1, 1], s[2, 2]]
+                _scales[index] = value
+                s = Scale.from_factors(_scales)
+            elif label == "Rotation":
+                _angles = r.euler_angles()
+                _angles[index] = value
+                r = Rotation.from_euler_angles(_angles)
+            obj.transformation = t * r * s
+            obj.update()
+
+        def get_transformation(label: str, index: int) -> float:  # type: ignore
+            if obj.transformation is None:
+                obj.transformation = Transformation()
+            s, _, r, t, _ = obj.transformation.decomposed()
+            if label == "Translation":
+                return t.translation_vector[index]  # type: ignore
+            elif label == "Scale":
+                return s[index, index]
+            elif label == "Rotation":
+                return r.euler_angles()[index]
```

Comment:
does this need to be here?

Reply:
that is the efficient way i can find out. any suggestions? :-)